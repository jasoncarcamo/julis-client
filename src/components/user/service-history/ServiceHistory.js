import React from 'react';
import TokenService from '../../../services/TokenService';
import {format as formatDate, getTime} from 'date-fns'
import {Link} from 'react-router-dom';
import './servicehistory.css';


export default class ServiceHistory extends React.Component{
       
    constructor(props){
        super(props);
        this.state = {
            services: [],
            service_type: '',
            comments: '',
            day: '',
            best_time_reached: '',
            date_created: '',
            price: '',
            confirm: false
        }
    }

    componentDidMount(){
        
        fetch('https://fathomless-eyrie-65525.herokuapp.com/user/service', {            
            headers: {
                'authorization': `bearer ${TokenService.getAuthToken()}`
            }
        })
            .then( res => res.json())
            .then( resData => this.setState({services: resData.services}));
    }


    handleCancelService = (e)=>{
        e.preventDefault()

        fetch(`https://fathomless-eyrie-65525.herokuapp.com/user/service`, {
            method: 'DELETE',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            },
            body: JSON.stringify({id: e.target.id})

        })
            .then( res => {
                this.componentDidMount();
            })

        
    }

    
    render(){
        
        return(
            <section id="service_header">
                <ul>
                    {this.state.services.map( service => (
                    <li key={service.id}>
                       
                        <header className="hi">Date set for {formatDate(getTime(service.date_modified), 'MMM Do YYYY ')}{service.best_time_reached}</header>
                        <p>{service.service_type}</p>
                        <p>{service.comments}</p>
                        <Link to={`/user/editservice?id=${service.id}`}>Edit</Link>  
                        {this.state.confirm ? (
                        <>
                            <button onClick={this.handleCancelService} className={service.id}>confirm</button> 
                            
                            <button className="cancel_confirm" onClick={()=> this.setState({confirm: false})}>Cancel</button>
                        </>) : <button type="button" className="cancel_confirm" onClick={() => this.setState({confirm: true})} >Cancel clean up</button>}                         
                    </li>))}
                </ul>
            </section>
        )
    }
}