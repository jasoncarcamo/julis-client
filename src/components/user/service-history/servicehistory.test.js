import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter} from 'react-router-dom';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ServiceHistory from './ServiceHistory';

Enzyme.configure({ adapter: new Adapter()});


describe('<ServiceHistory/>', ()=>{
    let serviceArray, wrapper;
    beforeEach(()=>{
        serviceArray =[
            {
                best_time_reached: "10:30 PM",
                comments: "Hello",
                date_created: "2019-07-27T08:45:54.567Z",
                date_modified: "2019-07-27T08:45:54.536Z",
                day: "2019-07-27T04:45:41.893Z",
                id: 105,
                price: "40",
                service_type: "Refridgerator, Windows",
                user_id: "40488d95-ce3b",
            },
            {
                best_time_reached: "10:15 AM",
                comments: "Hello",
                date_created: "2019-07-26T03:57:29.282Z",
                date_modified: "2019-07-27T08:54:28.676Z",
                day: "2019-07-27T04:54:16.611Z",
                id: 104,
                price: "20",
                service_type: "Windows",
                user_id: "40488d95-ce3b",
            },
            {
                best_time_reached: "12:30 AM",
                comments: "It worked.",
                date_created: "2019-07-27T08:58:17.773Z",
                date_modified: "2019-07-27T08:58:17.723Z",
                day: "2019-07-27T04:58:01.665Z",
                id: 106,
                price: "70",
                service_type: "Windows,  Refridgerator, Walls",
                user_id: "40488d95-ce3b",
            }
        ]

        wrapper = shallow(<ServiceHistory/>);

        wrapper.setState({services: serviceArray});
    })
    it('renders component', ()=>{
        const div = document.createElement('div');

        ReactDOM.render(<BrowserRouter><ServiceHistory/></BrowserRouter>, div);

        ReactDOM.unmountComponentAtNode(div);
    });

    it('Has the list of services to display', ()=>{

        expect(wrapper.find('li'));
    });
})