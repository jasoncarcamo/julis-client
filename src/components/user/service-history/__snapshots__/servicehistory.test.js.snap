// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceHistory/> snapshot oof ServiceComponent 1`] = `
<section
  id="service_header"
>
  <ul />
</section>
`;
